geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/UpdatePropagationDUnitTest.java

```diff
@@ -305,6 +360,32 @@ private void verifyUpdates() {
     });
   }
 
+  private void doPuts(int entries) throws Exception {
+    Region r1 = getCache().getRegion(REGION_NAME);
+    assertNotNull(r1);
+    for (int i = 0; i < entries; i++) {
+      try {
+        r1.put("" + i, "" + i);
+      } catch (Exception e) {
+        // ignore
```

Comment:
It doesn't seem like you should see any exceptions from put since your client put is being sent to server1 which never goes down. So it seems like you should remove this try/catch and just throw any unexpected exception to the caller of doPuts. Do you see exceptions here?

Reply:
We will get here org.apache.geode.cache.client.ServerConnectivityException: Pool unexpected socket timed out on client...

Reply:
If the client times out the put (which is why you get this exception) then the operation is still in progress on the servers. While it is still in progress because it is waiting for server2 to leave the membership view it will have already done the put on server1 but not yet distributed it to server3. You need to wait for these in progress ops to finish before doing your validation. It seems like this would still be a flaky test even with your new LIMIT feature.